core/controllers/library_test.py

```diff
@@ -194,14 +197,138 @@ def test_library_handler_for_created_explorations(self):
         }, response_dict['activity_list'][0])
 
 
+class LibraryIndexHandlerTests(test_utils.GenericTestBase):
+
+    def setUp(self):
+        super(LibraryIndexHandlerTests, self).setUp()
+        self.signup(self.VIEWER_EMAIL, self.VIEWER_USERNAME)
+        self.viewer_id = self.get_user_id_from_email(self.VIEWER_EMAIL)
+
+    def test_library_index_handler_for_user_preferred_language(self):
+        """Test the access to user's preferences."""
+        self.login(self.VIEWER_EMAIL)
+        response = self.get_html_response('/preferences')
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        # Change the user's preferred language to de.
+        self.put_json(
+            '/preferenceshandler/data',
+            {'update_type': 'preferred_language_codes', 'data': ['de']},
+            csrf_token=csrf_token)
+        response_dict = self.get_json(feconf.LIBRARY_INDEX_DATA_URL)
+        self.assertDictContainsSubset({
+            'activity_summary_dicts_by_category': [],
+            'preferred_language_codes': ['de'],
+        }, response_dict)
+
+    def test_library_index_handler_for_top_rated(self):
+        """Test the handler for top rated scenario."""
+        response_dict = self.get_json(feconf.LIBRARY_INDEX_DATA_URL)
+        self.assertDictContainsSubset({
+            'activity_summary_dicts_by_category': [],
+            'preferred_language_codes': ['en'],
+        }, response_dict)
+
+        # Load a demo.
+        exp_services.load_demo('0')
+        rating_services.assign_rating_to_exploration('user', '0', 2)
+        response_dict = self.get_json(feconf.LIBRARY_INDEX_DATA_URL)
+        self.assertEqual(
+            len(response_dict['activity_summary_dicts_by_category']), 1)
+        self.assertDictContainsSubset({
+            'preferred_language_codes': ['en'],
+        }, response_dict)
+        self.assertDictContainsSubset({
+            'categories': [],
+            'header_i18n_id': (
+                feconf.LIBRARY_CATEGORY_TOP_RATED_EXPLORATIONS),
+            'has_full_results_page': True,
+            'full_results_url': feconf.LIBRARY_TOP_RATED_URL,
+            'protractor_id': 'top-rated',
+        }, response_dict['activity_summary_dicts_by_category'][0])
+        activity_summary = response_dict['activity_summary_dicts_by_category'][
+            0]['activity_summary_dicts']
+        self.assertEqual(
+            len(activity_summary), 1)
+        self.assertDictContainsSubset({
+            'id': '0',
+            'category': 'Welcome',
+            'title': 'Welcome to Oppia!',
+            'language_code': 'en',
+            'objective': 'become familiar with Oppia\'s capabilities',
+            'status': rights_manager.ACTIVITY_STATUS_PUBLIC,
+        }, activity_summary[0])
+
+    def test_library_index_handler_for_featured(self):
+        """Test the handler for featured scenario."""
```

Comment:
`Test the handler for featured explorations.`

Reply:
[Fixed in this commit](https://github.com/MLA98/oppia/commit/48297ff460178c1aa4a5aab68081a8ef63e85081)